src/views/table_cell.ts

```diff
@@ -0,0 +1,92 @@
+/*!
+ * © 2019 Atypon Systems LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { schema } from '@manuscripts/transform'
+import { DOMSerializer, ResolvedPos } from 'prosemirror-model'
+
+import {
+  ContextMenu,
+  ContextMenuButton,
+} from '../components/views/TableCellContextMenu'
+import BlockView from './block_view'
+import { createNodeView } from './creators'
+import { EditableBlockProps } from './editable_block'
+import ReactSubView from './ReactSubView'
+
+export class TableCellView extends BlockView<EditableBlockProps> {
+  public contentDOM: HTMLElement
+
+  public initialise = () => {
+    if (!this.node.type.spec.toDOM) {
+      return
+    }
+    const outputSpec = this.node.type.spec.toDOM(this.node)
+    const { dom } = DOMSerializer.renderSpec(document, outputSpec)
+    this.dom = dom as HTMLElement
+    this.contentDOM = document.createElement('span')
+    this.dom.appendChild(this.contentDOM)
+
+    if (isInTable(this.view.state.doc.resolve(this.getPos()))) {
+      this.createContextMenu()
```

Comment:
Can we create the context menu on demand instead of doing on init? I'm worried of having too many react components rendered. We can the dots icon there always, and render the dropdown when it's clicked on.

Reply:
updated

Reply:
The react component is still being created for every cell. We should have the dots icon as plain HTML and only create the component on click.

Reply:
Done